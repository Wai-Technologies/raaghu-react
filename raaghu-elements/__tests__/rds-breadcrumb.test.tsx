// import React from "react";
// import { render, fireEvent, screen } from "@testing-library/react";
// import "@testing-library/jest-dom";
// import RdsBreadcrumb, { BreadcrumbProps } from "../src/rds-breadcrumb/rds-breadcrumb";

// jest.mock('react-lottie-player', () => ({
//     __esModule: true,
//     default: jest.fn(),
//   }));

// const defaultProps: BreadcrumbProps = {
//     breadItems: [
//         {
//             id: 1,
//             label: "Home",
//             route: "/home",
//             active: true,
//             disabled: false,
//         },
//         {
//             id: 2,
//             label: "Page 1",
//             route: "/page1",
//             active: false,
//             disabled: false,
//         },
//         {
//             id: 3,
//             label: "Page 2",
//             route: "/page2",
//             active: false,
//             disabled: false,
//         },
//     ],
// };

// describe("RdsBreadcrumb", () => {
//     it("renders correctly with default props", () => {
//         const { container } = render(<RdsBreadcrumb {...defaultProps} />);
//         expect(container).toMatchSnapshot();
//     });

//     //   it("should toggle active state when clicked", () => {
//     //     render(<RdsBreadcrumb {...defaultProps} />);
//     //     const page1 = screen.getByText("Page1");
//     //     const page2 = screen.getByText("Page2");

//     //     fireEvent.click(page1);
//     //     expect(page1).toHaveClass("active");
//     //     expect(page2).not.toHaveClass("active");

// //     fireEvent.click(page2);
// //     expect(page2).toHaveClass("active");
// //     expect(page1).not.toHaveClass("active");
// //   });
// });
