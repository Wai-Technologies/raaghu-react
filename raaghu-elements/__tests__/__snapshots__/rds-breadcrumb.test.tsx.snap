// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RdsBreadcrumb renders correctly with default props 1`] = `
<div>
  <nav
    aria-label="breadcrumb"
  >
    <ol
      class="breadcrumb m-0"
    >
      <li
        class="breadcrumb-item  active "
        id="breaditem1"
      >
        <a
          aria-disabled="true"
          class="text-decoration-none"
        >
          Home
        </a>
      </li>
      <li
        class="breadcrumb-items   "
        id="breadcrumbItem+2"
      >
        <span>
          <span
            class="cursor-pointer undefined"
            role="img"
          >
            <svg
              id="a"
              style="height: 8px; width: 14px; fill: none; stroke: currentColor;"
              viewBox="0 0 12 21"
            >
              <path
                class="icon"
                d="M.5,20.5L11.236,11.038c.337-.279,.354-.746,.038-1.044-.012-.011-.025-.022-.038-.033L.5,.5"
              />
            </svg>
          </span>
        </span>
        <a
          aria-disabled="true"
        >
          Page 1
        </a>
      </li>
      <li
        class="breadcrumb-items   text-primary"
        id="breadcrumbItem+3"
      >
        <span>
          <span
            class="cursor-pointer undefined"
            role="img"
          >
            <svg
              id="a"
              style="height: 8px; width: 14px; fill: none; stroke: currentColor;"
              viewBox="0 0 12 21"
            >
              <path
                class="icon"
                d="M.5,20.5L11.236,11.038c.337-.279,.354-.746,.038-1.044-.012-.011-.025-.022-.038-.033L.5,.5"
              />
            </svg>
          </span>
        </span>
        <a
          aria-disabled="true"
        >
          Page 2
        </a>
      </li>
    </ol>
  </nav>
</div>
`;
